### ISSUE TITLE
Binding in Entry should not modify the content while user is typing

### ISSUE THREAD

**Issue by fpabl0** [fyne-io/fyne#1849]
### Describe the bug:

Currently, binding a variable for an Entry updates the current text while user is typing and it is quite annoying.

I think the solution could be as easier as adding an early return if entry is focused in the binding listener:
```
	e.textListener = binding.NewDataListener(func() {
                // >>>> added 👇
		if e.focused { 
			return
		}
                // <<<<<<<<
		val, err := data.Get()
		if err != nil {
			convertErr = err
			e.SetValidationError(err)
			return
		}
```

### To Reproduce:
Steps to reproduce the behaviour:
1. Run fyne_demo
2. Go to "Data Binding" tab.
3. Try to edit the float value in the Entry by removing the "." from the left (pressing `Supr`).

### Screenshots:

https://user-images.githubusercontent.com/12239342/105461640-80f12a00-5c5b-11eb-8193-f8f2d0f12847.mp4

### Device (please complete the following information):
 - **OS:** MacOS
 - **Version:** 10.15.4 Catalina 
 - **Go version:** 1.15.6 
 - **Fyne version:** develop @ 547e07423

**Comment by andydotxyz** [comment]
Part of the design is that data updates immediately, so the general concept should remain.
I think there is an issue here however - but perhaps it relates to validation.
The reason these are "jumping" looks to be because the data is being re-formatted to validate then sent back to Entry.
If that is the case then perhaps it should not update if validation is failing or some similar, narrower focus fix?

**Comment by fpabl0** [comment]
> Part of the design is that data updates immediately, so the general concept should remain.

Yes, I know that, and it will still updates immediately but not in the element that is focused, because user expect to write something and see just what he/she writes, otherwise it is confusing.

> I think there is an issue here however - but perhaps it relates to validation.

I don't think it is related specifically to the validation. I think it is because data binding overwrite user input with the value returned by `data.Get()` and inside it occurs conversions that change the original string.

**Comment by andydotxyz** [comment]
Ah yes I think I misunderstood the original request - for Entry to ignore the updated value could indeed make sense.
I think when I said "validation" I should have said "validation and conversion", your point is a good one.
